File: Backend/src/main/java/com/copernic/backend/Backend/entity/enums/EstatReserva.java
package com.copernic.backend.Backend.entity.enums;

/**
 * Enum que representa els diferents estats d'una reserva dins del sistema.
 * Controla el cicle de vida d'una reserva des de la seva creació fins a la seva finalització o cancel·lació.
 */
public enum EstatReserva {

    /**
     * La reserva ha estat creada i assignada a una recompensa però encara no ha estat recollida.
     */
    RESERVADA,

    /**
     * La reserva ha estat completada amb èxit i la recompensa ha estat recollida per l'usuari.
     */
    RECOLLIDA,

    /**
     * La reserva ha estat cancel·lada abans de ser recollida.
     */
    CANCELADA,

    /**
     * La reserva ha estat formalment assignada a l'usuari per a la seva recollida.
     */
    ASSIGNADA,

    /**
     * La reserva ha estat desassignada, deixant la recompensa disponible de nou.
     */
    DESASSIGNADA,

    /**
     * La reserva ha caducat en no ser recollida dins del termini establert.
     */
    CADUCADA,

    /**
     * La reserva està pendent de recollir però ja ha estat gestionada.
     */
    PER_RECOLLIR
}

File: Frontend/app/src/main/java/cat/copernic/amayo/frontend/recompensaManagment/model/EstatReserva.kt
package cat.copernic.amayo.frontend.recompensaManagment.model

/**
 * Enum que representa els diferents estats possibles d'una reserva.
 *
 * Permet identificar en quina fase del procés es troba una reserva dins del sistema.
 */
enum class EstatReserva {

    /** La reserva ha estat creada i està pendent de ser assignada o recollida. */
    RESERVADA,

    /** La reserva ha estat recollida per l'usuari. */
    RECOLLIDA,

    /** La reserva ha estat cancel·lada. */
    CANCELADA,

    /** La reserva ha estat assignada a una recompensa i usuari. */
    ASSIGNADA,

    /** La reserva ha estat desassignada i retorna a l'estat inicial. */
    DESASSIGNADA,

    /** La reserva ha caducat per no haver estat recollida a temps. */
    CADUCADA,

    /** La reserva està pendent de ser recollida pel seu destinatari. */
    PER_RECOLLIR
}
